### DISCUSSION TITLE
Does Prisma suffer of the problem of useless order by on some ORM prefetches ?

### DISCUSSION THREAD

**Discussion by LLyaudet** [prisma/prisma#23523] [Q&A]
### Question

Hello,

Recently I found a problem with Django and submitted a patch in a PR:
https://code.djangoproject.com/ticket/35309
https://github.com/django/django/pull/17984
I also developped a work-around for old versions of Django:
https://github.com/LLyaudet/django-monkey-patches/blob/main/src/django_monkey_patches/django__orm__prefetch_without_useless_order_by.py

I don't know yet Prisma details.
The idea is just that some models can have a default ordering (I assume this functionality is present in Prisma also).
And that some ORM may still add this ordering in the SQL query whilst it is not needed because only a single object will be prefetched per instance of the original result set.
On very big prefetches I noticed a gain of 10 - 15 %,
since the rest of the DB query is mainly an index scan.
#ClimateChangeBrake

Can someone please dump the queries of such prefetches and see if the bug is present in Prisma ?
If so submitting a PR should not be too hard.
I will try to help.
But right now, I start by warning the frameworks/ORM I am aware of :)

Best regards,
      Laurent Lyaudet

**Comment by ludralph** [comment]
Hello @LLyaudet 👋,

Thank you for reaching out and sharing the work you've done to optimize Django's performance. 

At Prisma, we always aim to ensure our queries are as efficient as possible, but your specific concern about the default ordering might need closer inspection.

You could enable query [logs](https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/logging) to generate the SQL generated when using [orderBy](https://www.prisma.io/docs/orm/reference/prisma-client-reference#orderby)

As you have better context about the default ordering, you might be able to verify it by inspecting the generated SQL and check if there is any bug, or any improvements that can be made.

**Reply by LLyaudet** [reply]
Hello @ludralph :)
Thanks for your answer. I took a deeper look at Prisma documentation.
I could not find anything like prefetch or default ordering for a model.
Is there anything like that in Prisma ?
From what I have see, Prisma should not be concerned by the bug :)
Best regards,
    Laurent Lyaudet

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
